mlos_core/mlos_core/optimizers/optimizer.py

```diff
@@ -78,34 +80,40 @@ def register(self, configurations: pd.DataFrame, scores: pd.DataFrame,
             Dataframe of configurations / parameters. The columns are parameter names and the rows are the configurations.
         scores : pd.DataFrame
             Scores from running the configurations. The index is the same as the index of the configurations.
-
         context : pd.DataFrame
-            Not Yet Implemented.
+            Not implemented yet.
+        metadata : pd.DataFrame
+            Implementaton depends on instance.
         """
         # Do some input validation.
-        assert set(scores.columns) == set(self._optimization_targets), \
-            "Mismatched optimization targets."
+        if type(self._optimization_targets) is str:
```

Comment:
```suggestion
        assert self._optimization_targets, "Missing or invalid optimization targets"
        if type(self._optimization_targets) is str:
```

Reply:
Also assert not empty (see also comment above about accepting `None`)

Reply:
I don't think this makes sense given my comment above

Reply:
OK, but separate PR for that one please